inst/stan/functions/construct_spatial_rt.stan

```diff
@@ -0,0 +1,37 @@
+/**
+  * Generates the subpopulation Rt trajectories for a spatially correlated model
+  * given all the random components.
+  * @param log_state_rt "State" level unadjusted Rt, on log scale.
+  * @param spatial_deviation_ar_coeff Coefficient for AR(1) temporal correlation on
+  * subpopulation deviations.
+  * @param spatial_deviation_init Initial vector of the spatial deviation
+  * @param spatial_deviation_noise_matrix matrix with random vectors of noise
+  * for spatial devaiation process.
+  * @return A matrix for unadjusted Rt where rows are subpopulations
+  * and columns are time points.
+  */
+matrix construct_spatial_rt(vector log_state_rt,
+                            real spatial_deviation_ar_coeff,
+                            vector spatial_deviation_init,
+                            matrix spatial_deviation_noise_matrix) {
+
+
+  // presets
+  int n_time = cols(spatial_deviation_noise_matrix);
+  int n_sites = rows(spatial_deviation_noise_matrix);
+  matrix[n_sites,n_time] log_site_rt;
+  matrix[n_sites,n_time] spatial_deviation;
+
+  spatial_deviation[,1] = spatial_deviation_init;
+  for (t_i in 2:n_time) {
+    spatial_deviation[,t_i] = (spatial_deviation_ar_coeff*spatial_deviation[,t_i-1])
```

Comment:
This looks great! Really nice job with passing in the deviation matrix and the initial error -- my only question is in the way we are naming things, which is a bit of semantics! 

$$
\begin{pmatrix}
        \log R_1^u(t)\\
	\vdots\\
	\log R_{n_{k}}^u(t)\\
\end{pmatrix} = \log R^u(t)
\begin{pmatrix}
	1\\
	\vdots\\
	1\\
\end{pmatrix} + \vec{\delta}(t).
$$

Where $\vec{\delta}(t) = \left(\delta_1(t),...,\delta_{n_k}(t)\right)^{T}$.  

We also have that $\vec{\delta}(t) = \varphi_{R(t)}\vec{\delta}(t - 1) + \vec{\epsilon}$, where $\vec{\epsilon}$ $\sim$ $\text{Normal}_{n_k}(0,\mathbf{\Sigma\_{\epsilon}})$  

In terms of this nomenclature, I see $ \varphi_{R(t)}$ which is your `spatial_deviation_ar_coeff` as still representing a temporal autocorrelation on the previous time step's deviation from the state mean -- that may or may not be informed by any spatial (e.g. distance or adjacency) matrix. For example, we could pass in a `spatial_deviation_noise_matrix` that is drawn from a correlation matrix that has no explicit distance/spatial dependence -- it just describes how correlated the sites are to one another. 

To give a very concrete example, say we had 3 sites in a state, 2 from big cities with a transit line between them and 1 from a rural town nearby one of the cities. In theory, we might expect that the spatial information is not the relevant component here, but that mobility patterns/flux in and out of the two cities make them more correlated in infection dynamics than the 1 rural area. 

I think I myself have been getting confused when we are actually referring to things that are spatially dependent e.g. functions of distance $d$ vs functions of the subpopulations ($k$)

I think this is a bit of semantics so feel free to leave as is in this PR and make a new issue! I think

Reply:
e.g. $\mathbf{x} \sim \mathcal{N}(\mathbf{0}, \mathbf{I})$ where $\mathbf{x} = (x_1, \ldots, x_n)$ and $1, \ldots, n$ are indices for areas would be spatial random effects but unstructured.

Reply:
Edited @kaitejohnson comment to fix a typo that was messing up $\TeX$ rendering